lib/slurm_squeue_client.rb

```diff
@@ -61,7 +61,49 @@ def sinfo
     
     s.success? ? @sinfo = o : raise(CommandFailed, e)
   end
-  
+
+  # Parse and return total number of GPU nodes in a SLURM cluster.
+  # @return [Integer] number of GPU nodes
+  def gpu_nodes
+    return @available_gpu_nodes if defined?(@available_gpu_nodes)
+
+    Open3.pipeline_rw "sinfo -N -h --states=allocated,idle --Format='nodehost,gres:100,gresused:100'", 'uniq', 'grep gpu:v', 'wc -l' do |stdin, stdout|
+      stdin.write stdout
+      stdin.close
+      @available_gpu_nodes = stdout.read.to_i
+    end
+  end
+
+  # Return number of jobs requesting GPU nodes
+  # @return [Integer] number of GPU nodes with status mixed (some CPUs allocated)
+  def gpu_nodes_active
+    return @gpu_nodes_active if defined?(@gpu_nodes_active)
+
+    Open3.pipeline_rw "sinfo -a -h --states=mixed --Node --Format='nodehost,gres:100,gresused:100'", 'uniq', 'grep gpu:v', 'wc -l' do |stdin, stdout|
```

Comment:
Because you are not parsing the actual GPU usage, and just rather getting GPU nodes that are busy, no need to query `gresused:100`.

Also mixed is wrong, that will miss `allocated`.   Should be `--states=mixed,allocated`

Reply:
When using `--states=mixed,allocated` the result is the same. `allocated` seems to include `mixed` where as just `mixed` does not include `allocated` nodes.